gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/DefaultTraversalStrategies.java

```diff
@@ -25,25 +25,23 @@
 
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 
 /**
  * @author Marko A. Rodriguez (http://markorodriguez.com)
  */
 public class DefaultTraversalStrategies implements TraversalStrategies {
 
-    protected List<TraversalStrategy<?>> traversalStrategies = new ArrayList<>();
+    protected Set<TraversalStrategy<?>> traversalStrategies = new LinkedHashSet<>();
 
     @Override
     @SuppressWarnings({"unchecked", "varargs"})
     public TraversalStrategies addStrategies(final TraversalStrategy<?>... strategies) {
-        final List<TraversalStrategy<?>> concurrent = new ArrayList<>(this.traversalStrategies);
         for (final TraversalStrategy<?> addStrategy : strategies) {
-            for (final TraversalStrategy<?> currentStrategy : concurrent) {
-                if (addStrategy.getClass().equals(currentStrategy.getClass()))
-                    this.traversalStrategies.remove(currentStrategy);
-            }
+            this.traversalStrategies.remove(addStrategy);
```

Comment:
Do we really need this loop? In a `Set` we can just keep adding strategies, nothing would happen if a strategy was already added.

Reply:
I don't know, I'm just keeping the same functionality but with a different data structure.

Reply:
Yea, I just tried to get rid of the loop, but that doesn't work because at some point we rely on a replacement behavior.